bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/LinkImpl.java

```diff
@@ -0,0 +1,187 @@
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.internal.models.v1.formsportal;
+
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ChildResource;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.forms.core.components.internal.models.v1.AbstractComponentImpl;
+import com.adobe.cq.forms.core.components.models.formsportal.Link;
+
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = { Link.class, ComponentExporter.class },
+    resourceType = LinkImpl.RESOURCE_TYPE)
+@Exporter(
+    name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class LinkImpl extends AbstractComponentImpl implements Link {
+    // ToDo: Documentation of each method once properties and model is reviewed and final
+    public static final String RESOURCE_TYPE = "core/fd/components/formsportal/link/v1/link";
+    public static final String QUERY_PARAMS_PATH = "queryParams";
+    private static final String PN_PARAM_KEY = "key";
+    private static final String PN_PARAM_VALUE = "value";
+    public static final Logger logger = LoggerFactory.getLogger(LinkImpl.class);
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Inject
+    private String linkText;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Inject
+    private String linkTooltip;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Inject
+
+    private String assetType;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Inject
+    private String adaptiveformPath;
+
+    @ChildResource(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Named(QUERY_PARAMS_PATH)
+    private List<Resource> paramsResourceList;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+    private Map<String, String> queryParamsMap;
+
+    private String linkTarget;
+
+    @Override
+    public String getRenderUrl() {
```

Comment:
Why is this public ? if you need access in test cases, you could do it via reflection for now.

Reply:
As discussed exposing this in json and removing query parameters from the interface